blockstore/src/main/java/brooklyn/location/blockstore/openstack/OpenstackNewVolumeCustomizer.java

```diff
@@ -0,0 +1,119 @@
+package brooklyn.location.blockstore.openstack;
+
+import brooklyn.location.blockstore.BlockDeviceOptions;
+import brooklyn.location.blockstore.FilesystemOptions;
+import brooklyn.location.blockstore.api.MountedBlockDevice;
+import brooklyn.location.blockstore.api.VolumeManager;
+import com.google.common.base.Optional;
+import com.google.common.collect.Maps;
+import org.apache.brooklyn.location.jclouds.BasicJcloudsLocationCustomizer;
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
+import org.apache.brooklyn.location.jclouds.JcloudsMachineLocation;
+import org.jclouds.compute.ComputeService;
+import org.jclouds.compute.domain.NodeMetadata;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Customizer that can be used for attaching additional disk on provisioning time for OpenStack.<br><br>
+ *
+ * Below is shown an example:
+ *
+ * <pre>
+ *   provisioning.properties:
+ *     customizers:
+ *     - $brooklyn:object:
+ *         type: brooklyn.location.blockstore.openstack.OpenstackNewVolumeCustomizer
+ *         object.fields:
+ *           volumes:
+ *           - blockDevice:
+ *               sizeInGb: 3
+ *               deviceSuffix: 'b'
+ *               deleteOnTermination: false
+ *               tags:
+ *                 brooklyn: br-example-val-test-1
+ *            filesystem:
+ *              mountPoint: /mount/brooklyn/b
+ *              filesystemType: ext3
+ * </pre>
+ *
+ * Important notice is that KVM is configured as the default hypervisor for OpenStack which means that the defined device name will be of type /dev/vd*.
+ * This means that the device suffix must be set as the next letter in alphabetical order from the existing device names on the VM.
+ */
+
+public class OpenstackNewVolumeCustomizer extends BasicJcloudsLocationCustomizer {
+
+    private static final OpenstackVolumeManager openstackVolumeManager = new OpenstackVolumeManager();
+
+    private Map<BlockDeviceOptions, FilesystemOptions> volumes;
+
+    protected MountedBlockDevice mountedBlockDevice;
```

Comment:
Don't store this in the instance (same for the other implementations). Instead add a new method in `NewVolumeCustomizer` which returns `MountedBlockDevice` which does all the work. Call the new method from the effector & from the `customize` method (where the code lives atm).